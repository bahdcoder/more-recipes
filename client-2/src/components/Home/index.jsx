import React from 'react';
import LeadButtons from './components/LeadButtons';

class Home extends React.Component {
  render() {
    return (
      <div id="jumbotron" className="jumbotron text-center">
        <h1 className="display-3 mb-5">
          <img src={`${process.env.PUBLIC_URL}/img/logo.png`} className="jumbotron-logo-img mr-2" />
        </h1>
        <p className="lead jumbotron-title display-4 wow bounceInUp" style={{color: 'white'}}>Making everyday cooking fun !</p>
        <br />
        <p className="lead">
          {/* The home button 
                  If the user is authenticated, let him rather see a create recipe button
              */}
          <LeadButtons />
          {/* End of the home button */}
        </p>
      </div>
    );
  }
}

export default Home;
