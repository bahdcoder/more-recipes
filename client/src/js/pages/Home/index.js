import React from 'react';
import Footer from '../../components/Footer';
import Navbar from '../../components/Navbar';
import RecipeCard from '../../components/RecipeCard';

import bannerImage1 from '../../../assets/img/banner-1.jpg';
import bannerImage2 from '../../../assets/img/banner-2.jpg';
import bannerImage3 from '../../../assets/img/banner-3.jpg';

export default class Home extends React.Component {
  render() {
    return (
      <div>
        {/* The navigation bar begin */}
        <Navbar/>
        {/* The navigation bar ending */}
        {/* The Jumbotron Area */}
        <div id="jumbotron" className="jumbotron text-center">
          <h1 className="display-3 mb-5">
            <img src="../../../assets/img/logo.png" className="jumbotron-logo-img mr-2" alt />
          </h1>
          <p className="lead jumbotron-title display-4 wow bounceInUp" data-wow-duration="2s" style={{color: 'white'}}>Making everyday cooking fun !</p>
          <br />
          <p className="lead">
            {/* The home button 
                    If the user is authenticated, let him rather see a create recipe button
                */}
            <a href="login.html" className="btn btn-primary btn-lg mr-2" role="button" data-toggle="modal" data-target="#exampleModal">Sign in</a>                        
            <a href="login.html" className="btn btn-primary btn-lg" role="button" data-toggle="modal" data-target="#registerModal">Join now</a>
            {/* End of the home button */}
          </p>
        </div>
        {/* End of the jumbotron area */}
        {/* Top rated recipes section */}
        <div className="container-fluid px-5 my-5">
          <h1 className="display-4 text-center my-5 wow fadeInDown" style={{padding: '30px 0px'}}>
            <i className="ion ion-star mr-3" />
            Top Rated Recipes</h1>
          <br />
          <div className="card-deck wow fadeIn" data-wow-duration="4s">
            <RecipeCard/>
            <RecipeCard/>
            <RecipeCard/>            
          </div>
        </div>
        {/* End of top rated recipes section */}
        {/* Top rated chefs section */}
        <div className="container px-5 my-5">
          <h1 className="display-4 text-center my-5 wow fadeInDown" style={{padding: '30px 0px'}}>
            <i className="ion ion-person-stalker mr-3" />
            Most Popular Chefs</h1>
          <br />
          <div className="row wow fadeIn" data-wow-duration="4s">
            <div className="col-lg-3 col-md-4 col-sm-6 col-xs-12">
              <div className="hovereffect">
                <img className="img-fluid img-responsive" src="http://i.pravatar.cc/300" alt />
                <div className="overlay">
                  <p className="my-auto">
                    <a href="profile.html">KATI FRANTZ</a>
                  </p>
                </div>
              </div>
            </div>
            <div className="col-lg-3 col-md-4 col-sm-6 col-xs-12">
              <div className="hovereffect">
                <img className="img-fluid img-responsive" src="http://i.pravatar.cc/300" alt />
                <div className="overlay">
                  <p className="my-auto">
                    <a href="profile.html">SELINA RYANS</a>
                  </p>
                </div>
              </div>
            </div>
            <div className="col-lg-3 col-md-4 col-sm-6 col-xs-12">
              <div className="hovereffect">
                <img className="img-fluid img-responsive" src="http://i.pravatar.cc/300" alt />
                <div className="overlay">
                  <p className="my-auto">
                    <a href="profile.html">ELLEN MICHEALS</a>
                  </p>
                </div>
              </div>
            </div>
            <div className="col-lg-3 col-md-4 col-sm-6 col-xs-12">
              <div className="hovereffect">
                <img className="img-fluid img-responsive" src="http://i.pravatar.cc/300" alt />
                <div className="overlay">
                  <p className="my-auto">
                    <a href="profile.html">CLAUS LEBRONE</a>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        {/* End of top rated chefs section */}
        {/* Most favorited recipes section */}
        <div className="container-fluid px-5 my-5">
          <h1 className="display-4 text-center my-5 wow fadeInDown" style={{padding: '30px 0px'}}>
            <i className="ion ion-heart mr-4" />
            Most favorited Recipes</h1>
          <br />
          <div className="card-deck wow fadeIn" data-wow-duration="4s">
            <RecipeCard/>
            <RecipeCard/>
            <RecipeCard/>            
          </div>
        </div>
        {/* End of top rated recipes section */}
        <div className="modal fade" id="exampleModal" tabIndex={-1} role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
          <div className="modal-dialog" role="document" style={{paddingTop: 130}}>
            <div className="modal-content">
              <div className="modal-body text-center">
                <div className="row justify-content-center">
                  <div className="col-10">
                    <form>
                      <h3 className="text-center mb-3 mt-3">Login</h3>
                      <div className="form-group">
                        <input type="text" className="form-control" id="staticEmail" placeholder="email@example.com" />                                    
                      </div>  
                      <div className="form-group">
                        <input type="password" className="form-control" id="inputPassword" placeholder="Password" />                                    
                      </div>
                      <div className="form-group">
                        <button className="btn mb-3 btn-primary form-control" type="submit">Login</button>   
                        <span className="mt-5 h6 mr-3"><a href role="button" data-toggle="modal" data-target="#registerModal" data-dismiss="modal">No account ? Register</a></span>    
                        <span className="mt-5 ml-2 h6"><a href>Forgot your password?</a></span>                                 
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        {/*Begin page footer */}
        <Footer/>
        {/* End page footer */}
        <div className="modal fade" id="registerModal" tabIndex={-1} role="dialog" aria-labelledby="registerModalLabel" aria-hidden="true">
          <div className="modal-dialog" role="document" style={{paddingTop: 70}}>
            <div className="modal-content">
              <div className="modal-body text-center">
                <div className="row justify-content-center">
                  <div className="col-10">
                    <form>
                      <h3 className="text-center mb-3 mt-3">Register</h3>
                      <div className="form-group">
                        <input type="text" className="form-control" placeholder="First name" />                                    
                      </div>  
                      <div className="form-group">
                        <input type="text" className="form-control" placeholder="Last name" />                                    
                      </div>  
                      <div className="form-group">
                        <input type="email" className="form-control" id="staticEmail" placeholder="email@example.com" />                                    
                      </div>  
                      <div className="form-group">
                        <input type="password" className="form-control" id="inputPassword" placeholder="Password" />                                    
                      </div>
                      <div className="form-group">
                        <input type="password" className="form-control" id="inputPassword" placeholder="Confirm Password" />                                    
                      </div>
                      <div className="form-group text-center">
                        <button className="btn mb-3 btn-primary form-control" type="submit">Register</button>   
                        <span className="mt-5 h6 mr-3"><a href role="button" data-toggle="modal" data-target="#exampleModal" data-dismiss="modal">Have an account? Login</a></span>                              
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
