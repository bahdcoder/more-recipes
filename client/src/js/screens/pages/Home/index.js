import React from 'react';
import { Link } from 'react-router';
import Footer from '../../components/Footer';
import Navbar from '../../components/Navbar';
import SignIn from '../../components/auth/SignIn';
import RecipeCard from '../../components/RecipeCard';

import bannerImage1 from '../../../../assets/img/banner-1.jpg';

export default class Home extends React.Component {
  render() {
    let homeButtons = (
      <span>
        <Link to="/auth/login" className="btn btn-primary btn-lg mr-2" role="button">Sign in</Link>                        
        <Link to="/auth/register" className="btn btn-primary btn-lg" role="button" >Join now </Link>
      </span>  
    );

    if (this.props.authUser) {
      homeButtons = (
        <span>
          <Link to="/recipes/create" className="btn btn-primary btn-lg mr-2"> Create recipe </Link>                        
          <Link to="/" className="btn btn-primary btn-lg" role="button"> View your recipes </Link>
        </span>  
      );
    } 
    
    return (
      <div>
        {/* The navigation bar begin */}
        <Navbar {...this.props}/>
        {/* The navigation bar ending */}
        {/* The Jumbotron Area */}
        <div id="jumbotron" className="jumbotron text-center">
          <h1 className="display-3 mb-5">
            <img src="../../../assets/img/logo.png" className="jumbotron-logo-img mr-2" />
          </h1>
          <p className="lead jumbotron-title display-4 wow bounceInUp" style={{color: 'white'}}>Making everyday cooking fun !</p>
          <br />
          <p className="lead">
            {/* The home button 
                    If the user is authenticated, let him rather see a create recipe button
                */}
            {homeButtons}
            {/* End of the home button */}
          </p>
        </div>
        {/* End of the jumbotron area */}
        {/* Top rated recipes section */}
        <div className="container-fluid px-5 my-5">
          <h1 className="display-4 text-center my-5 wow fadeInDown" style={{padding: '30px 0px'}}>
            <i className="ion ion-star mr-3" />
            Top Rated Recipes</h1>
          <br />
          <div className="card-deck wow fadeIn">
            <RecipeCard/>     
            <RecipeCard/>     
            <RecipeCard/>                        
          </div>
        </div>
        {/* End of top rated recipes section */}
        {/* Top rated chefs section */}
        <div className="container px-5 my-5">
          <h1 className="display-4 text-center my-5 wow fadeInDown" style={{padding: '30px 0px'}}>
            <i className="ion ion-person-stalker mr-3" />
            Most Popular Chefs</h1>
          <br />
          <div className="row wow fadeIn">
            <div className="col-lg-3 col-md-4 col-sm-6 col-xs-12">
              <div className="hovereffect">
                <img className="img-fluid img-responsive" src="http://i.pravatar.cc/300" />
                <div className="overlay">
                  <p className="my-auto">
                    <a>KATI FRANTZ</a>
                  </p>
                </div>
              </div>
            </div>
            <div className="col-lg-3 col-md-4 col-sm-6 col-xs-12">
              <div className="hovereffect">
                <img className="img-fluid img-responsive" src="http://i.pravatar.cc/300"/>
                <div className="overlay">
                  <p className="my-auto">
                    <a >SELINA RYANS</a>
                  </p>
                </div>
              </div>
            </div>
            <div className="col-lg-3 col-md-4 col-sm-6 col-xs-12">
              <div className="hovereffect">
                <img className="img-fluid img-responsive" src="http://i.pravatar.cc/300" />
                <div className="overlay">
                  <p className="my-auto">
                    <a>ELLEN MICHEALS</a>
                  </p>
                </div>
              </div>
            </div>
            <div className="col-lg-3 col-md-4 col-sm-6 col-xs-12">
              <div className="hovereffect">
                <img className="img-fluid img-responsive" src="http://i.pravatar.cc/300" />
                <div className="overlay">
                  <p className="my-auto">
                    <a>CLAUS LEBRONE</a>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        {/* End of top rated chefs section */}
        {/* Most favorited recipes section */}
        <div className="container-fluid px-5 my-5">
          <h1 className="display-4 text-center my-5 wow fadeInDown" style={{padding: '30px 0px'}}>
            <i className="ion ion-heart mr-4" />
            Most favorited Recipes</h1>
          <br />
          <div className="card-deck wow fadeIn">
            <RecipeCard/>
            <RecipeCard/>
            <RecipeCard/>            
          </div>
        </div>
        {/* End of top rated recipes section */}
        <SignIn data={this.props}/>
        {/*Begin page footer */}
        <Footer/>
        {/* End page footer */}
        <div className="modal fade" id="registerModal" tabIndex={-1} role="dialog" aria-labelledby="registerModalLabel" aria-hidden="true">
          <div className="modal-dialog" role="document" style={{paddingTop: 70}}>
            <div className="modal-content">
              <div className="modal-body text-center">
                <div className="row justify-content-center">
                  <div className="col-10">
                    <form>
                      <h3 className="text-center mb-3 mt-3">Register</h3>
                      <div className="form-group">
                        <input type="text" className="form-control" placeholder="First name" />                                    
                      </div>  
                      <div className="form-group">
                        <input type="text" className="form-control" placeholder="Last name" />                                    
                      </div>  
                      <div className="form-group">
                        <input type="email" className="form-control" id="staticEmail" placeholder="email@example.com" />                                    
                      </div>  
                      <div className="form-group">
                        <input type="password" className="form-control" id="inputPassword" placeholder="Password" />                                    
                      </div>
                      <div className="form-group">
                        <input type="password" className="form-control" id="inputPassword" placeholder="Confirm Password" />                                    
                      </div>
                      <div className="form-group text-center">
                        <button className="btn mb-3 btn-primary form-control" type="submit">Register</button>   
                        <span className="mt-5 h6 mr-3"><a role="button" data-toggle="modal" data-target="#exampleModal" data-dismiss="modal">Have an account? Login</a></span>                              
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
