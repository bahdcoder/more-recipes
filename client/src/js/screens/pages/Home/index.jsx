import React from 'react';
import PropTypes from 'prop-types';

import Footer from '../../components/Footer';
import Navbar from '../../components/Navbar';
import logo from './../../../../assets/img/logo.png';
import RecipeCard from '../../components/RecipeCard';
import HomeButtons from './components/HomeButtons';
import RecipeCardLoader from '../../components/RecipeCardLoader';

/**
 * The home page component
 */
class Home extends React.Component {
  static propTypes = {
    getHomePageData: PropTypes.func
  };

  static defaultProps = {
    getHomePageData: () => {}
  };
  /**
   * The home page props
   * @param {obj} props props
   */
  constructor(props) {
    super(props);

    this.state = {
      loading: true
    };
  }
  /**
   * Trigger when component mounts
   * @returns {null} null
   */
  async componentWillMount() {
    await this.props.getHomePageData();
    this.setState({
      loading: false
    });
  }

  /**
   * Render jsx for this component
   * @returns {obj} jsx
   */
  render() {
    let latestRecipes = [];

    if (this.state.loading) {
      latestRecipes = (
        <div className="row">
          <div className="col-md-4">
            <RecipeCardLoader />
          </div>
          <div className="col-md-4">
            <RecipeCardLoader />
          </div>
          <div className="col-md-4">
            <RecipeCardLoader />
          </div>
        </div>
      );
    } else {
      let numberOnPage = 3;
      const sortedLatestRecipes = this.props.recipes.sort((recipe1, recipe2) =>
        recipe1.createdAt < recipe2.createdAt);
      if (sortedLatestRecipes.length > 5) {
        numberOnPage = 6;
      }
      const trimmedListOfLatestRecipes = sortedLatestRecipes.slice(0, numberOnPage);
      latestRecipes = trimmedListOfLatestRecipes.map(recipe => ((
        <div className="col-md-4" key={recipe.id}>
          <RecipeCard recipe={recipe} {...this.props} />
        </div>
      )));
    }

    let mostFavoritedRecipes = [];
    if (this.state.loading) {
      mostFavoritedRecipes = (
        <div className="row">
          <div className="col-md-4">
            <RecipeCardLoader />
          </div>
          <div className="col-md-4">
            <RecipeCardLoader />
          </div>
          <div className="col-md-4">
            <RecipeCardLoader />
          </div>
        </div>
      );
    } else {
      let numberOnPage = 3;
      const sortedMostFavoritedRecipes = this.props.recipes.sort((recipe1, recipe2) =>
        recipe1.favoritersIds.length < recipe2.favoritersIds.length);
      if (sortedMostFavoritedRecipes.length > 5) {
        numberOnPage = 6;
      }
      const trimmedListOfMostFavoritedRecipes = sortedMostFavoritedRecipes.slice(0, numberOnPage);
      mostFavoritedRecipes = trimmedListOfMostFavoritedRecipes.map(recipe => ((
        <div className="col-md-4" key={recipe.id}>
          <RecipeCard recipe={recipe} {...this.props} />
        </div>
      )));
    }

    return (
      <div>
        {/* The navigation bar begin */}
        <Navbar {...this.props} />
        {/* The navigation bar ending */}
        {/* The Jumbotron Area */}
        <div id="jumbotron" className="jumbotron text-center">
          <h1 className="display-3 mb-5">
            <img src={logo} className="jumbotron-logo-img mr-2" alt="" />
          </h1>
          <p className="lead jumbotron-title display-4 wow bounceInUp" style={{ color: 'white' }}>Making everyday cooking fun !</p>
          <br />
          <p className="lead">
            <HomeButtons
              authUser={this.props.authUser}
              isAuthenticated={this.props.authUser !== undefined && this.props.authUser !== null}
            />
            {/* End of the home button */}
          </p>
        </div>
        {/* End of the jumbotron area */}
        {/* Top rated recipes section */}
        <div className="container-fluid px-5 my-5">
          <h1 className="display-5 text-center my-5 wow fadeInDown" style={{ padding: '30px 0px' }}>
            <i className="ion ion-star mr-3" />
            Lastest Recipes
          </h1>
          <br />
          <div className="card-deck wow fadeIn">
            {latestRecipes}
          </div>
        </div>
        {/* End of top rated recipes section */}
        {/* Most favorited recipes section */}
        <div className="container-fluid px-5 my-5">
          <h1 className="display-5 text-center my-5 wow fadeInDown" style={{ padding: '30px 0px' }}>
            <i className="ion ion-heart mr-4" />
            Most favorited Recipes
          </h1>
          <br />
          <div className="card-deck wow fadeIn">
            {mostFavoritedRecipes}
          </div>
        </div>
        <br /><br /><br />
        {/* End of top rated recipes section */}

        {/* Begin page footer */}
        <Footer />
        {/* End page footer */}
      </div>
    );
  }
}

export default Home;
