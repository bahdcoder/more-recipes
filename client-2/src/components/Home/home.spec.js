import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from './index';

it('Should home component render correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Home/>
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});