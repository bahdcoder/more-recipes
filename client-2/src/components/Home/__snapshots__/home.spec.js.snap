// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should home component render correctly 1`] = `
<div
  className="jumbotron text-center"
  id="jumbotron"
>
  <h1
    className="display-3 mb-5"
  >
    <img
      className="jumbotron-logo-img mr-2"
      src="/img/logo.png"
    />
  </h1>
  <p
    className="lead jumbotron-title display-4 wow bounceInUp"
    style={
      Object {
        "color": "white",
      }
    }
  >
    Making everyday cooking fun !
  </p>
  <br />
  <p
    className="lead"
  >
    <span>
      <span>
        <a
          className="btn btn-primary btn-lg mr-2"
          href="/recipes/create"
          onClick={[Function]}
          role="button"
        >
          Create recipe
        </a>
        <a
          className="btn btn-primary btn-lg"
          href="/user/recipes"
          onClick={[Function]}
          role="button"
        >
          Manage your recipes
        </a>
      </span>
    </span>
  </p>
</div>
`;
